internal/server/iigointernal/executive.go

```diff
@@ -141,11 +143,13 @@ func (e *executive) replyAllocationRequest(commonPool shared.Resources, aliveIsl
 		if !e.incurServiceCharge(actionCost.ReplyAllocationRequestsActionCost) {
 			return errors.Errorf("Insufficient Budget in common Pool: replyAllocationRequest")
 		}
-		for _, islandID := range aliveIslands {
-			d := shared.CommunicationContent{T: shared.CommunicationInt, IntegerData: int(allocationMapReturn.ResourceMap[islandID])}
-			data := make(map[shared.CommunicationFieldName]shared.CommunicationContent)
-			data[shared.AllocationAmount] = d
-			communicateWithIslands(islandID, shared.TeamIDs[e.PresidentID], data)
+		for islandID, resourceAmount := range allocationMapReturn.ResourceMap {
+			if Contains(aliveIslands, islandID) {
+				d := shared.CommunicationContent{T: shared.CommunicationInt, IntegerData: int(resourceAmount)}
```

Comment:
If someone ask for an allocation and the president does not respond they are given no update of that. This is fine, it should just be added as a comment in the CommonPoolRequest function in the baseclient

Reply:
Could we possibly give these agents like a message with 0 allocations, I feel like it might mess up teams agents if they get nothing at all.